rust/pact_ffi/src/models/message.rs

```diff
@@ -160,6 +165,7 @@ ffi_fn! {
     /// a NULL message and a missing message body.
     fn pactffi_message_get_contents(message: *const Message) -> *const c_char {
         let message = as_ref!(message);
+        let message = block_on(generate_message(&message, &GeneratorTestMode::Consumer, &hashmap!{}, &vec![], &hashmap!{}));
```

Comment:
`block_on` will block the current thread until the future has run to completion. If this function is called concurrently (i.e. from two different threads from the other side of the FFI at the same time), I think it will cause a deadlock.

Reply:
No, I think this is ok as long as it is only called via FFI. If it is called from another FFI function that uses a Tokio runtime, it will block the runtime and cause a deadlock.

Reply:
Currently I'm only interested in getting generators work with `pactffi_message_reify`.

Should I close PRs for `pactffi_message_get_contents` (this PR) and `pactffi_message_metadata_iter_next` (#341) to lower the risk?

Reply:
No, I think it is good for now. As long as these are only called via FFI. I can't think of a reason why another FFI function would call this. They should call the underlying functions (i.e. `generate_message`) themselves.

Reply:
Maybe just add a line to the doc comment under the Safety section that the function must only be called via foreign languages and not Rust code, otherwise a deadlock could result.

Reply:
Can you help me add a commit to this PR? I will copy the comment to other methods

Reply:
The commits are in your repo, which I won't be able to do. Here is the comment:

```rust
/// # Safety
///
/// The returned string must be deleted with `pactffi_string_delete` and can outlive the message.
/// This function must only ever be called from a foreign language. Calling it from a Rust function
/// that has a Tokio runtime in its call stack can result in a deadlock.
```

Reply:
Thanks @rholshausen